craft_application/commands/lifecycle.py

```diff
@@ -183,6 +161,15 @@ def _run(
         """Run a lifecycle step command."""
         super()._run(parsed_args)
 
+        if self._use_provider(parsed_args):
+            fetch_service_policy = getattr(parsed_args, "fetch_service_policy", None)
+            if fetch_service_policy:
+                self._services.get("fetch").set_policy(
+                    fetch_service_policy  # type: ignore[reportArgumentType]
+                )
+            self._run_manager_for_build_plan(fetch_service_policy)
```

Comment:
If I'm understanding this PR correctly, there are now two ways to run managed commands: through `Application.run_managed()` (the old way) and through the ProviderService (the new way). Do we need to keep the old way around?

Reply:
I want to remove the old way, but I didn't want to inflate this PR specifically with that.